cmd/export.go

```diff
@@ -0,0 +1,86 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package cmd defines command line utilities for ghpc
+package cmd
+
+import (
+	"fmt"
+	"hpc-toolkit/pkg/shell"
+	"log"
+	"os"
+	"path"
+
+	"github.com/spf13/cobra"
+)
+
+func init() {
+	metadataFlag := "metadata"
+	artifactsFlag := "artifacts"
+	exportCmd.Flags().StringVarP(&artifactsDir, artifactsFlag, "a", "", "Alternative artifacts output directory (automatically configured if unset)")
+	exportCmd.Flags().StringVarP(&metadataFile, metadataFlag, "m", "", "Deployment metadata YAML file (automatically configured if unset)")
+	exportCmd.MarkFlagDirname(artifactsFlag)
+	exportCmd.MarkFlagFilename(metadataFlag, "yaml", "yml")
+	rootCmd.AddCommand(exportCmd)
+}
+
+const defaultMetadataFile string = "../.ghpc/deployment_metadata.yaml"
```

Comment:
Just to confirm, this is just the expanded blueprint? Part of me wants to name it as such, but naming it as you have it might discourage people from messing with it, where naming it something like `expanded_blueprint` might make people think they could move it.

Reply:
It is not yet the expanded blueprint, but will be. When #1205 is merged the deployment metadata file will continue to be written without modification, so these are not in conflict in the branch sense or functionality sense. However, #1205 will make it much easier to get the information directly from the expanded blueprint without potential mismatch. Additionally, it will enable proper/elegant support for writing inputs to Packer modules that will not be present in the follow-up to this PR.

I could imagine:

- changing the flag to `--blueprint-metadata` (or similar) if we are worried about exposing the name now on develop branch.
- putting big giant "DO_NOT_DELETE_THIS_DIRECTORY" type file into the ".ghpc" directory when writing the deployment, along with the expanded blueprint when we write it